File: FLEX.Data.Oracle/DbLoggerBase.vb
Public MustInherit Class DbLoggerBase

   Protected Const MaxApplicationNameLength As Integer = 30
   Protected Const MaxCodeUnitLength As Integer = 100
   Protected Const MaxFunctionLength As Integer = 100
   Protected Const MaxShortMessageLength As Integer = 400
   Protected Const MaxLongMessageLength As Integer = 4000
   Protected Const MaxContextLength As Integer = 400
    Protected Const MaxKeyLength As Integer = 100
    Protected Const MaxValueLength As Integer = 400
    Protected Const MaxArgumentCount As Integer = 10

   ''' <summary>
   ''' 
   ''' </summary>
   ''' <param name="exception"></param>
   ''' <returns></returns>
   ''' <remarks></remarks>
   Protected Shared Function FindInnermostException(exception As Exception) As Exception
      While exception.InnerException IsNot Nothing
         exception = exception.InnerException
      End While
      Return exception
   End Function

End Class

File: Caravan.DataAccess/Core/DbLoggerBase.cs
using System;

namespace FLEX.DataAccess.Core
{
    public abstract class DbLoggerBase
    {
        protected const int MaxApplicationNameLength = 30;
        protected const int MaxCodeUnitLength = 100;
        protected const int MaxFunctionLength = 100;
        protected const int MaxShortMessageLength = 400;
        protected const int MaxLongMessageLength = 4000;
        protected const int MaxContextLength = 400;
        protected const int MaxKeyLength = 100;
        protected const int MaxValueLength = 400;
        protected const int MaxArgumentCount = 10;

        /// <summary>
        /// </summary>
        /// <param name="exception"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        protected static Exception FindInnermostException(Exception exception)
        {
            while (exception.InnerException != null)
            {
                exception = exception.InnerException;
            }
            return exception;
        }
    }
}